File: DesignPatternsExercises/ruby/strategy_design_pattern/solution/db_extraction.rb
require_relative 'extraction'

class DBExtraction < Extraction
  def extract
    # __________(5)__________
    open
    puts 'DB Data type is getting extracted by reading rows and columns...'
    # __________(6)__________
    close
  end

  def open
    puts 'Opening DB file...'
  end

  def close
    puts 'Closing DB file...'
  end
end

File: DesignPatternsExercises/java/StrategyDesignPattern/Solution/DBExtraction.java
class DBExtraction implements Extraction {
    @Override
    public void extract() {
        // __________(7)__________;
        open();
        System.out.println("DB Data type is getting extracted by reading rows and columns...");
        // __________(8)__________;
        close();
    }

    @Override
    public void open() {
        System.out.println("Opening DB file...");
    }

    @Override
    public void close() {
        System.out.println("Closing DB file...");
    }
}
